testing/trino-product-tests/src/main/java/io/trino/tests/product/hive/TestHiveCoercionOnUnPartitionedTable.java

```diff
@@ -0,0 +1,122 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.tests.product.hive;
+
+import com.google.common.collect.ImmutableMap;
+import io.trino.tempto.Requirement;
+import io.trino.tempto.RequirementsProvider;
+import io.trino.tempto.Requires;
+import io.trino.tempto.configuration.Configuration;
+import io.trino.tempto.fulfillment.table.MutableTableRequirement;
+import io.trino.tempto.fulfillment.table.hive.HiveTableDefinition;
+import org.testng.annotations.Test;
+
+import java.util.Map;
+import java.util.Optional;
+
+import static io.trino.tempto.fulfillment.table.MutableTableRequirement.State.CREATED;
+import static io.trino.tests.product.TestGroups.HIVE_COERCION;
+import static io.trino.tests.product.TestGroups.JDBC;
+import static java.lang.String.format;
+import static java.util.Locale.ENGLISH;
+
+public class TestHiveCoercionOnUnPartitionedTable
+        extends BaseTestHiveCoercion
+{
+    public static final HiveTableDefinition HIVE_COERCION_ORC = tableDefinitionBuilder("ORC", Optional.empty(), Optional.empty())
+            .setNoData()
+            .build();
+
+    private static HiveTableDefinition.HiveTableDefinitionBuilder tableDefinitionBuilder(String fileFormat, Optional<String> recommendTableName, Optional<String> rowFormat)
+    {
+        String tableName = format("%s_hive_coercion_unpartitioned", recommendTableName.orElse(fileFormat).toLowerCase(ENGLISH));
```

Comment:
recommendTableName -> tableNamePrefix

why  is it optional? can our test pick test table name more deterministically ?